### PR TITLE
Improve headers of the number field section in toc

### PR DESCRIPTION
**Pull request by kwankyu**
- Decapitalize titles in the number field section
- Make numerous fixes to make headers look good

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/rings/number_field/maps.py [hunk 1]
```diff
@@ -1,8 +1,8 @@
 r"""
 Structure maps for number fields
 
-Provides isomorphisms between relative and absolute presentations, to and from
-vector spaces, name changing maps, etc.
+Thid module provides isomorphisms between relative and absolute presentations,
```
Comment (mkoeppe) [inline src/sage/rings/number_field/maps.py]:
```suggestion
This module provides isomorphisms between relative and absolute presentations,
```

File: src/sage/rings/number_field/S_unit_solver.py [hunk 2]
```diff
@@ -1,24 +1,10 @@
 # sage.doctest: needs sage.rings.number_field sage.rings.padics
 r"""
-Solve S-unit equation x + y = 1
+Solve `S`-unit equation `x + y = 1`
```
Comment (mkoeppe) [inline src/sage/rings/number_field/S_unit_solver.py]: This title stands out grammatically. Perhaps "Solver for the ..."

File: src/sage/rings/number_field/maps.py (hunk 1 above)
Comment (kwankyu) [reply src/sage/rings/number_field/maps.py]: Done. Thanks.

File: src/sage/rings/number_field/S_unit_solver.py (hunk 2 above)
Comment (kwankyu) [reply src/sage/rings/number_field/S_unit_solver.py]: Thanks. Done.

File: src/sage/rings/number_field/S_unit_solver.py [hunk 3]
```diff
@@ -1,10 +1,10 @@
 # sage.doctest: needs sage.rings.number_field sage.rings.padics
 r"""
-Solve `S`-unit equation `x + y = 1`
+Solve for the `S`-unit equation `x + y = 1`
```
Comment (mkoeppe) [inline src/sage/rings/number_field/S_unit_solver.py]: What I meant is that this section title is a verb phrase, whereas the others are noun phrases

Comment (kwankyu) [reply src/sage/rings/number_field/S_unit_solver.py]: Ah, sorry. Right. Fixed.

**Review by mkoeppe — APPROVED** [review]
LGTM, this is a nice improvement for this section

**Comment by kwankyu** [comment]
Thanks!

**Pull request closed without merging**